new-components/scanners/gosec/internal/sarif/sarif.go

```diff
@@ -0,0 +1,139 @@
+package sarif
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+)
+
+const (
+	// GoSecOutFormatJson tells the parser to unmarshal the raw data using json umarshaler.
+	GoSecOutFormatJson GoSecOutFormat = "json"
+)
+
+type (
+	// GosecOut represents the sarif output returned by running https://github.com/securego/gosec.
+	GosecOut struct {
+		Issues []GoSecIssue `json:"Issues"`
+	}
+
+	// GoSecIssue maps a go sec issue.
+	GoSecIssue struct {
+		Severity     string   `json:"severity"`
+		Confidence   string   `json:"confidence"`
+		Cwe          GoSecCwe `json:"cwe"`
+		RuleID       string   `json:"rule_id"`
+		Details      string   `json:"details"`
+		File         string   `json:"file"`
+		Code         string   `json:"code"`
+		Line         string   `json:"line"`
+		Column       string   `json:"column"`
+		Nosec        bool     `json:"nosec"`
+		Suppressions any      `json:"suppressions"`
+	}
+
+	// GoSecCwe describes the gosec cwe metadata.
+	GoSecCwe struct {
+		ID  string `json:"id"`
+		URL string `json:"url"`
+	}
+
+	// GoSecOutFormat represents the possible raw data formats supplied to the parser.
+	GoSecOutFormat string
+
+	// Parser implements sarif parsing for gosec out.
+	Parser struct {
+		rawOutFilePath string
+		rawOut         string
+		format         GoSecOutFormat
+	}
+)
+
+func (g GoSecOutFormat) String() string {
+	return string(g)
+}
+
+// NewParserFromEnv returns a new Parsed by doing environment variables lookup.
+func NewParserFromEnv() (*Parser, error) {
+	parseFormat, err := env.GetOrDefault(
+		"GOSEC_OUT_FORMAT",
```

Comment:
we should only support a single output format per tool, if it supports something interoperable then that's the preference otherwise we deal with their json but please, supporting absolutely everything is boiling the ocean